tests/src/test/java/alluxio/client/rest/TestCase.java

```diff
@@ -27,78 +27,45 @@
 import java.io.OutputStream;
 import java.net.HttpURLConnection;
 import java.net.URL;
+import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 import javax.annotation.concurrent.NotThreadSafe;
+import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
 /**
  * Represents a REST API test case.
  */
 @NotThreadSafe
 public final class TestCase {
-  private String mHostname;
-  private int mPort;
-  private String mEndpoint;
-  private Map<String, String> mParameters;
-  private String mMethod;
-  private Object mExpectedResult;
-  private TestCaseOptions mOptions;
-  private String mPrefix;
+  private final String mHostname;
+  private final int mPort;
+  private final String mBaseUri;
+  private final String mEndpoint;
+  private final Map<String, String> mParameters;
+  private final String mMethod;
+  private final TestCaseOptions mOptions;
 
   /**
    * Creates a new instance of {@link TestCase}.
-   *
    * @param hostname the hostname to use
    * @param port the port to use
+   * @param baseUri the base URI of the REST server
    * @param endpoint the endpoint to use
    * @param parameters the parameters to use
    * @param method the method to use
-   * @param expectedResult the expected result to use
-   */
-  public TestCase(String hostname, int port, String endpoint, Map<String, String> parameters,
```

Comment:
This looks like a useful constructor to keep, why are we deleting this?

Reply:
I made passing the expected result a mandatory parameter to the method where it was being used, as it is not necessary for all instances of `TestCase`.

Previously:
```
public void run() throws Exception {
```

is now:
```
public void runAndCheckResult(Object expectedResult) throws Exception {
```

Reply:
oh, i agree with removing the `expectedResult`. I just think the not having to provide `TestCaseOptions` is nice to have.

Reply:
Oh, my bad. I misunderstood.

Currently `TestCaseOptions.defaults()` looks like
```
  /**
   * @return the default {@link TestCaseOptions}
   */
  public static TestCaseOptions defaults() {
    return new TestCaseOptions();
  }
  private TestCaseOptions() {
    mBody = null;
    mInputStream = null;
    mPrettyPrint = false;
    mContentType = JSON_CONTENT_TYPE;
    mMD5 = null;
    mAuthorization = null;
  }
```

I'd rather have people explicitly choose to use the "default" rather than be given an implicit one in order to (try to) force developers to look at their options.
- I think the fact that the default content type is JSON is something which could bite developers in the butt (it did for me)

---
Otherwise if we want to hide the defaults from developers `public static TestCaseOptions defaults()` shouldn't even exist in that case. It could just be set in the constructor(s) of `TestCase`.